unstructured/partition/pdf.py

```diff
@@ -292,6 +285,36 @@ def _partition_pdf_or_image_local(
     if languages is None:
         languages = ["eng"]
 
+    if (
+        not pdf_text_extractable
+        and not is_image
+        and hi_res_reorientation_strategy not in [None, False, "", ReorientationStrategy.NONE]
+    ):
+        with tempfile.NamedTemporaryFile(mode="w") as tmpFile:
+            reorient_file_or_data(filename, file, hi_res_reorientation_strategy, tmpFile.name)
+            return _partition_pdf_or_image_local(
+                filename=tmpFile.name,  # the reoriented file
+                file=None,
+                is_image=is_image,
+                infer_table_structure=infer_table_structure,
+                include_page_breaks=include_page_breaks,
+                languages=languages,
+                ocr_mode=ocr_mode,
+                model_name=model_name,
+                hi_res_model_name=hi_res_model_name,
+                pdf_image_dpi=pdf_image_dpi,
+                metadata_last_modified=metadata_last_modified,
+                pdf_text_extractable=pdf_text_extractable,
+                extract_images_in_pdf=extract_images_in_pdf,
+                extract_image_block_types=extract_image_block_types,
+                extract_image_block_output_dir=extract_image_block_output_dir,
+                extract_image_block_to_payload=extract_image_block_to_payload,
+                analysis=analysis,
+                analyzed_image_output_dir_path=analyzed_image_output_dir_path,
+                hi_res_reorientation_strategy="",  # already reoriented
+                **kwargs,
+            )
+
```

Comment:
Can we do this within `partition_pdf_or_image()` after line [536](https://github.com/Unstructured-IO/unstructured/blob/main/unstructured/partition/pdf.py#L536)?

Reply:
I need clarification - line 536 might now refer to a different line than at the time of writing the above comment.

Reply:
It now refers to line [528](https://github.com/Unstructured-IO/unstructured/blob/main/unstructured/partition/pdf.py#L528) , before calling the `_partition_pdf_or_image_local()` within `partition_pdf_or_image()`

Reply:
Newest commit moves the logic to partition_pdf_or_image and removes the reorientation_strategy argument from _partition_pdf_or_image_local outright.

I'm not sure I like it more from an in-code perspective, but the data flow itself indeed feels more elegant now.